android/src/main/java/me/carda/awesome_notifications/background/AwesomeNotificationMessagingBackgroundExecutor.java

```diff
@@ -0,0 +1,321 @@
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, @NonNull Result result) {
+    Log.d(TAG,"onMethodCall");
+    String method = call.method;
+    try {
+      if (method.equals(Definitions.CHANNEL_METHOD_INIT_BACKGROUND)) {
+        // This message is sent by the background method channel as soon as the background isolate
+        // is running. From this point forward, the Android side of this plugin can send
+        // callback handles through the background method channel, and the Dart side will execute
+        // the Dart methods corresponding to those callback handles.
+        onInitialized();
+        result.success(true);
+      } else {
+        result.notImplemented();
+      }
+    } catch (PluginRegistrantException e) {
+      result.error("error", "Flutter Awesome notification error: " + e.getMessage(), null);
+    }
+  }
+
+  /**
+   * Starts running a background Dart isolate within a new {@link FlutterEngine} using a previously
+   * used entrypoint.
+   *
+   * <p>The isolate is configured as follows:
+   *
+   * <ul>
+   *   <li>Bundle Path: {@code io.flutter.view.FlutterMain.findAppBundlePath(context)}.
+   *   <li>Entrypoint: The Dart method used the last time this plugin was initialized in the
+   *       foreground.
+   *   <li>Run args: none.
+   * </ul>
+   *
+   * <p>Preconditions:
+   *
+   * <ul>
+   *   <li>The given callback must correspond to a registered Dart callback. If the handle does not
+   *       resolve to a Dart callback then this method does nothing.
+   *   <li>A static {@link #pluginRegistrantCallback} must exist, otherwise a {@link
+   *       PluginRegistrantException} will be thrown.
+   * </ul>
+   */
+  public void startBackgroundIsolate() {
+    if (isNotRunning()) {
+      long callbackHandle = getPluginCallbackHandle();
+      if (callbackHandle != 0) {
+        startBackgroundIsolate(callbackHandle, null);
+      }
+    }
+  }
+
+  /**
+   * Starts running a background Dart isolate within a new {@link FlutterEngine}.
+   *
+   * <p>The isolate is configured as follows:
+   *
+   * <ul>
+   *   <li>Bundle Path: {@code io.flutter.view.FlutterMain.findAppBundlePath(context)}.
+   *   <li>Entrypoint: The Dart method represented by {@code callbackHandle}.
+   *   <li>Run args: none.
+   * </ul>
+   *
+   * <p>Preconditions:
+   *
+   * <ul>
+   *   <li>The given {@code callbackHandle} must correspond to a registered Dart callback. If the
+   *       handle does not resolve to a Dart callback then this method does nothing.
+   *   <li>A static {@link #pluginRegistrantCallback} must exist, otherwise a {@link
+   *       PluginRegistrantException} will be thrown.
+   * </ul>
+   */
+  public void startBackgroundIsolate(long callbackHandle, FlutterShellArgs shellArgs) {
+    if (backgroundFlutterEngine != null) {
+      Log.e(TAG, "Background isolate already started.");
+      return;
+    }
+
+    Handler mainHandler = new Handler(Looper.getMainLooper());
```

Comment:
This main looper runs on main thread and gets in concurrency, competing with the core Flutter's engine and degrading the app performance.

Reply:
I think when the app is killed is not matter which isolate it will be,
what i did try to call this function only when the app is killed,but i did not had to time to figure how to do it,in your notfiticain life cycle when the app is kill and got push the value is in background.
In adittion why you want to call the function on other isolate,u cannot access to you current reesource(memory,db ,etc....).
For my opinion if someone want to active this functiom on other isolate it can do it by itslef,correct me if am i wrong.
But if you insists about it , i will try to assist u :)

Reply:
> I think when the app is killed is not matter which isolate it will be,

Totally agreed. The point is while the app is in background or foreground. Imagine to do a reply button to not bring the app to foreground.

If the app is in foreground and you press that notification button, a thread will be started with a new flutter engine, and it will get in concurrency with the main app and, on the user perspective, the app goes freeze some frames on each messsage sent. This can downgrade the users experience and, eventually, this plugin acceptance by the Flutter's comunity.

> what i did try to call this function only when the app is killed,but i did not had to time to figure how to do it,in your notfiticain life cycle when the app is kill and got push the value is in background.

Yes, i did the same strategy for `SilentMainThread` action. But this procedure can be terminated with the main app and some important information could be lost. This is why i believe is important to create two silent action types, one to run with this strategy and another to goes on background thread.

> In adittion why you want to call the function on other isolate,u cannot access to you current reesource(memory,db ,etc....).

Not totally true. The isolates only isolates the Flutter level. On native one, the static classes, network, storages and another resources are shared between the threads.

> For my opinion if someone want to active this functiom on other isolate it can do it by itslef,correct me if am i wrong.
> But if you insists about it , i will try to assist u :)

Im prettending to publish this first MVP with only SilentMainThread enabled. But further the background thread notification will be necessary as well. Now im replicating this isolates on iOS, but right now i have no clue how to run a flutter engine inside an backgroudn thread. If you can discovery it, it will helpe me a lot, for real.

Reply:
If it possible , i dont know how to do it here, to discuss about it in chat mode.

Reply:
Are you available right now to do a meeting call?
https://meet.google.com/ngz-ssth-zxk

Reply:
Not right now 😅,
But lets talk tomorrow

Reply:
No problem at all. I will try to do the MVP for iOS until there.

Reply:
send me your mail pls lets talk in private

Reply:
rafaelsetra@gmail.com